### ISSUE TITLE
Is bisq-on-docker a good choice for a bisq-beginner?

### ISSUE THREAD

**Issue by mitohund** [fflo/docker-bisq#1]
This is not really an issue, but since you don't have a discussions tab and there's no other link to contact you by, I thought I'd just ask here.

The title says it all, really: is this project stable and usable? Particularly for bisq beginners? On the bisq forums, nobody seems to have mentioned bisq-on-docker yet. So it would be difficult to impossible to get any support there. And particularly if one is new and runs into problems, it could be difficult to find out if the problem lies with Bisq itself or it running on docker.

**Comment by Conza88** [comment]
I am a Bisq vet (non-super technical) but Linux/docker noob. Do have a Truenas OS, and going to try set this up as VM - Ubuntu. 
To house.

**Comment by mitohund** [comment]
Well, for docker you don't need a full-blown VM. That's kind of the point of docker. Wait, just googled for a moment. Seems like Truenas is FreeBSD-based and docker of course is Linux. So you would possibly need a Linux-VM to run docker in, after all? Unless you go with Truenas Scale, which is Linux-based.

Just had a quick look at this discussion: https://www.reddit.com/r/DataHoarder/comments/n21rpe/freenas_truenas_core_and_docker/

Here's more info on Truenas Scale & docker: https://www.truenas.com/docs/scale/apps/docker/

Good luck and tell us how it went!

**Comment by fflo** [comment]
Hey guys,

I created this docker container, to be able to run Bisq instances 24/7 on any Docker compatible device.  Personally, I like the strong commitment of the Bisq community to privacy and decentralization using a peer-to-peer-based approach. That's awesome.

Maintaining this Docker project is my unofficial commitment to the project.

Using this docker image you can run your Bisq instance on any Docker compatible Internet-connected device, like for example a NAS or a hosted VPS. There is a free Bisq cellphone app available for both iOS and Android devices, which you can link to your Bisq instance. Having linked your Bisq app with your Bisq instance, you will receive push messages if your orders have been fulfilled, which is useful for time-optimized trading.

**Comment by mitohund** [comment]
Thanks for the response! I am also impressed by the Bisq community and how solid the project is.

From your response, it sounds like the bisq-on-docker experience is stable and smooth. I'll definitely give it a try then.

How come you haven't advertised it on [bisq.community](https://bisq.community/) yet? To me, this sounds like such a logical extension of the Bisq idea.

**Comment by Conza88** [comment]
Thanks gents. 

I realise its a bit of a long way of doing it; Truenas OS, then a VM for linux server lol, to then make a docker for Bisq. 
Basically my end goal was Bisq on server, amongst other things (Plex etc.). This is where I ended up. Would have gone to Truenas SCALE (linux) if I have known that key difference, and that this docker/bisq option was available beforehand. Wondering about upgrading. 

In any case, also have set up Portainer - https://www.youtube.com/watch?v=XBVjuwgz0Cg&t=612s 

Question is now - with this docker; how do I actually engage/use Bisq? lol ... any GUI that exists? 

"Remote Desktop (RDP) application" - steps or recommendations from this? I have what I thought were the IP and ports to access via browser but no dice :\ 

Running into probs with exposing the 3389 port, should just be able to pop that into browser and access yeah? 

_$ docker run -p 3389:3389 fflo/bisq-on-docker
docker: Error response from daemon: driver failed programming external connectivity on endpoint cranky_moser (cf5412d30bd0528a5a1f1508308988c57a78735e01772eb69): Bind for 0.0.0.0:3389 failed: port is already allocated.
ERRO[0000] error waiting for container: context canceled_

**Comment by mitohund** [comment]
I unfortunately didn't get around to try bisq-on-docker yet. So take the following as more general input. RDP stands for remote desktop protocol. So you would need a client to connect to the Bisq remote desktop. "Remote Desktop Connection" is the built-in client that's available in Windows. If you're on Linux or Mac, you would have to find a different client. But that shouldn't be difficult.

Hoping you'll get it to work and tell us about your progress!

**Comment by Conza88** [comment]
Yep. Got recommended xpra. I do believe I should just be able to ssh in locally from web browser though? 
Some network permission, issues with container, although fairly sure I exposed the port.

**Comment by mitohund** [comment]
Again, I haven't looked at this properly yet. But I believe Xpra is not an RDP client?

https://en.wikipedia.org/wiki/Comparison_of_remote_desktop_software

I think you need an RDP client. As in separate software you run on the client that wants to access bisq-on-docker. Not a web browser. And I'm not sure SSH has anything to do with it. You might be able to add it for security reasons. But for a first try, I wouldn't add that complication.

@fflo would know all this much better than me.

P.S.: I still can't believe we're the only two people interested in this.

**Comment by Conza88** [comment]
Thanks @fflo. Any chance of a walk through or something for the noobs here?

I need to setup on my truenas, a VM - linux sever running MATE desktop? It wouldn't work with Ubuntu live server? 
Think issue was is I never installed a GUI. 

EDIT:

WELL, geezus christ lol.. I'm in. Can remote in from windows 10 remote desktop to my Truenas OS -> VM -> of Linux Ubuntu server with minimalsetup/core, with MATE gui. Have setup portainer as well; but now I can't seem to find/see how to run Bisq when I'm in there?!

**Comment by mitohund** [comment]
You mean once you're in Ubuntu Mate? Click "Menu" at the top left, then "All", then "Bisq".

**Comment by Conza88** [comment]
Yeah, not there. :\ 
From indocker; but also - try downloading from inside mate terminal GUI and install etc, .deb - no dice.

**Comment by Conza88** [comment]
Ever see this? [https://github.com/bisq-network/docker](https://github.com/bisq-network/docker)

**Comment by mitohund** [comment]
Yes, I gathered it's a development environment. Not for users?

**Comment by fflo** [comment]
Question to the users of bisq-on-docker:

- What are your challenges installing and using bisq-on-docker?

- How do you like to use bisq-on-docker?
  * running on cloud VPS?
  * running on own NAS or other server hardware?
  * running on notebook / pc?
  * running as a software service for a monthly usage fee?
